pyccl/correlations.py

```diff
@@ -5,28 +5,29 @@
     'FFTLog' is fast using a fast Fourier transform.
     'Legendre' uses a sum over Legendre polynomials.
 """
+__all__ = ("CorrelationMethods", "CorrelationTypes", "correlation",
+           "correlation_3d", "correlation_multipole", "correlation_3dRsd",
+           "correlation_3dRsd_avgmu", "correlation_pi_sigma",)
 
-from . import ccllib as lib
-from . import constants as const
-from . import DEFAULT_POWER_SPECTRUM
-from .pyutils import check
-from .pk2d import parse_pk2d
-from .base import warn_api
-from .errors import CCLDeprecationWarning
-import numpy as np
 import warnings
 
-correlation_methods = {
-    'fftlog': const.CCL_CORR_FFTLOG,
-    'bessel': const.CCL_CORR_BESSEL,
-    'legendre': const.CCL_CORR_LGNDRE,
+import numpy as np
+
+from . import DEFAULT_POWER_SPECTRUM, check, lib
+from . import CCLDeprecationWarning, warn_api
+
+
+CorrelationMethods = {
+    'fftlog': lib.CCL_CORR_FFTLOG,
+    'bessel': lib.CCL_CORR_BESSEL,
+    'legendre': lib.CCL_CORR_LGNDRE,
```

Comment:
fixed

Reply:
Why add aliases rather than keep the old names? There's nothing wrong with the old names, and this just adds more clutter.

Reply:
See my comment regarding enums, in python enum types typically inherit from Enum and so they assume class-like naming. It is different to C, and this code must have been leftover from C.

Reply:
https://docs.python.org/3/library/enum.html

Reply:
this is a dictionary

Reply:
Yes, and it shouldn't be, because it behaves like an enum.
At some point, when these are brought to pure Python, they will inherit from `Enum` directly and work like the `NeutrinoMassSplits` in `neutrinos.py`.
Python users expect enums to be named like enums. Since I now expose them at the top level, I need to make this name change to be consistent.

Reply:
The alternative is to keep them private, but there is no need for that - people might find them useful because checking the keys reveals what is implemented in CCL. We will also refer to those objects in the docs. So they all have to have consistent naming.

Reply:
a) This is a dictionary, and there's no need to make them enums (and they're not enums).
b) Their current naming is perfectly fine and pep8 compliant.

Going out of our way to rename things that don't need renaming just makes the code more complex and users lives more difficult when transitioning to v3.

Reply:
Ok, all this is done now. I reverted all those changes, and simply defined some top-level enum types to point the users to from the docs. This is much better now I think.